### PR TITLE
docs: made doc files org clearer and augmented content

### PR DESCRIPTION
**Pull request by esciara**
Hi guys. Being a newbie, I struggled a fair bit getting a grip on `semantic-release` by reading the doc.

This is a first step to improving the doc: 
- renamed directories; 
- augmented a fair bit of content. (`usage` for the time being)

To be continued, but please merge this already (after reviewing it and after eventual corrections)

### CODE REVIEW

**Comment by gr2m** [comment]
Can you resolve the conflicts? Let me know if you need help with that :)

File: docs/01-usage/plugins.md [hunk 1]
```diff
@@ -80,11 +75,11 @@ Global plugin options can defined at the root of the **semantic-release** config
     "@semantic-release/commit-analyzer",
     "@semantic-release/release-notes-generator",
     ["@semantic-release/github", {
-      "assets": ["dist/**"]
+      "assets": ["dist/**"]                # plugin configuration
```
Comment (gr2m) [inline docs/01-usage/plugins.md]: Unfortunately, comments are not allowed in JSON. Adding them will make it harder to copy & paste the examples.

We could add a description below explaining the different bits of it?

File: docs/01-usage/configuration.md [hunk 2]
```diff
@@ -85,55 +112,101 @@ The git repository URL.
...
-### dryRun
+Example (`.releaserc` file content):
+```json
+{
+  "plugins": [
+    "@semantic-release/commit-analyzer",
+    "@semantic-release/release-notes-generator",
+    "@semantic-release-docker",
+    ["@semantic-release/exec", {
+      "verifyConditionsCmd": "./verify.sh"    # plugin configuration
```
Comment (gr2m) [inline docs/01-usage/configuration.md]: Unfortunately, comments are not allowed in JSON. Adding them will make it harder to copy & paste the examples.

We could add a description below explaining the different bits of it?

File: docs/01-usage/installation.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+## Global installation
+
+For other type of projects we recommend installing **semantic-release** directly in the CI environment, also with [npx](https://www.npmjs.com/package/npx):
+
+```bash
+$ npx semantic-release
+```
+
+> **Note:** For a global installation, it's recommended to specify the major **semantic-release** version to install (for example with with `npx semantic-release@15`, or `npm install -g semantic-release@15`). This way your build will not automatically use the next major **semantic-release** release that could possibly break your build. You will have to upgrade manually when a new major version is released.
```
Comment (gr2m) [inline docs/01-usage/installation.md]: I think I would no longer recommend to globally install semantic-release, the `npx` command works great and is supported in all actively maintained Node versions.

File: docs/01-usage/configuration.md (hunk 2 above)
Comment (esciara) [reply docs/01-usage/configuration.md]: Done in next push. Explanation already in the text.

File: docs/01-usage/plugins.md (hunk 1 above)
Comment (esciara) [reply docs/01-usage/plugins.md]: Done in next push. Explanation already in the text.

File: docs/01-usage/installation.md (hunk 3 above)
Comment (esciara) [reply docs/01-usage/installation.md]: Not sure I understand this part. Do you mean that I should remove the "Global installation" altogether? What is then the recommendation for CI environments?

Comment (gr2m) [reply docs/01-usage/installation.md]: Yes, remove `npm install -g semantic-release@15` altogether.

For CI environments, install semantic-release as a dev dependency and run `npx semantic-release`, which will use the locally installed version

Comment (esciara) [reply docs/01-usage/installation.md]: OK. Done in the next push.

**Comment by esciara** [comment]
> Can you resolve the conflicts? Let me know if you need help with that :)

Should I do a `git rebase` followed that a force push? Or is there a better way of doing it?

**Comment by gr2m** [comment]
> Should I do a `git rebase` followed by a forced push? Or is there a better way to do it?

That's how I'd do it :)

**Comment by esciara** [comment]
> > Should I do a `git rebase` followed by a forced push? Or is there a better way to do it?
> 
> That's how I'd do it :)

And `squash` or no `squash`?... to get a single commit... (can't remember if it affects the code review comments...)

**Comment by gr2m** [comment]
that works. We can always squash when merging the PR

**Comment by pvdlg** [comment]
Sorry that I didn't review this PR before it was merged...

There is some issues with this PR. Many changes in the docs are actually inaccurate and doesn't reflect how semantic-releae works and should be used.

I'm going to review those changes one by one and revert the appropriate changes. However, why were the directory named changed? What value does it brings to add a number in front of the directory? It seems to suggest the user there is an order for those sections while the intention is not have a specific order and let the users access each part depending on what they need.
The goal was initially to have an extensive and comprehensive doc without forcing users to read everything, for example a user who just want the common case (GitHub + npm + Travis) wouldn't have to read more than a few dozen of line of code. If that user want to customize further (via plugin or other) then they can read more doc.

**Pull request merged**